pkg/controller/operators/catalog/operator.go

```diff
@@ -1014,6 +1021,71 @@ func (o *Operator) createInstallPlan(namespace string, subs []*v1alpha1.Subscrip
 	return reference.GetReference(res)
 }
 
+func (o *Operator) checkBundleLookups(plan *v1alpha1.InstallPlan) (bool, error) {
+	//allFinished := true
+	for _, bundleLookup := range plan.Status.BundleLookups {
+		job, err := o.opClient.KubernetesInterface().BatchV1().Jobs(bundleLookup.BundleJob.Namespace).Get(bundleLookup.BundleJob.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		if len(job.Status.Conditions) > 0 && job.Status.Conditions[0].Type != batchv1.JobComplete {
+			// TODO: could write steps for each bundle image as ready
+			//allFinished = false
+			//continue
+			return false, nil
+		}
+		bundleLookup.BundleJob.Condition = job.Status.Conditions[0].Type
+		bundleLookup.BundleJob.CompletionTime = job.Status.CompletionTime
+
+		configmap, err := o.opClient.KubernetesInterface().CoreV1().ConfigMaps(bundleLookup.ConfigMapRef.Namespace).Get(bundleLookup.ConfigMapRef.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		// extract data from configmap and write to install plan
+		manifest, err := o.bundleLoader.Load(configmap)
+
+		// combine data from the bundle image into what's already known from the registry
+		bundleLookup.BundleFromRegistry.CsvName = manifest.Bundle.Name
+		bundleLookup.BundleFromRegistry.PackageName = manifest.Bundle.Package
+		bundleLookup.BundleFromRegistry.ChannelName = manifest.Bundle.Channel
+		_, jsonCSV, _, err := manifest.Bundle.Serialize()
+		if err != nil {
+			return false, fmt.Errorf("serialize failed: %s", err.Error())
+		}
+		bundleLookup.BundleFromRegistry.CsvJson = string(jsonCSV)
+		bundleLookup.BundleFromRegistry.Object, err = registry.BundleStringToObjectStrings(bundleLookup.BundleFromRegistry.CsvJson)
```

Comment:
I think this can just be `bundleLookup.BundleFromRegistry.Object = []string{string(jsonCSV)}`

Reply:
^ still this 😄

Reply:
Yeah good call. Anything to reduce potential errors here.